privacy-scaling-explorations/zkevm-circuits
# Testool various fixes/improvements

This PR contains a set of fixes/improvements to testool.

### Contents
- Catch panics
    - Panic `evm_unimplemented` message in order to capture the message and annotate the test with `Ignored` errorlevel  if does not fail for another reason
    - capture `circuit was not satisfied` panic messages and annotate them as `Fail` instead of `Panic` errorlevel.
- Tests with `expectException`
   - Fix to handle tests that are expected to fail (`exception`).
   - Add tests for it.
- Use assert_satisfied_par
  - Use `assert_satisfied_par()` when proving SuperCircuit
  - Use `CircuitTestBuilder` to test evm & state circuits (default usage). Now the `testool` is not able to capture the error messages in report (since are emitted to stderr), so we only know that fails the mockprover, but we can see the details when running individual tests with `--inspect`.
- Add `target = testool` to `debug!`s
- Virtual STOP
  - `STOP`s were removed initially in test code compilation when only virtual STOP  was available, now is implemented 
  - Update the code cache `codehash.txt` file to keep compiled code as is
- Fix `Dockerfile` to fix the compilation tools version
